sqlx-core/src/postgres/connection/stream.rs

```diff
@@ -41,6 +47,8 @@ impl PgStream {
         Ok(Self {
             inner,
             notifications: None,
+            parameter_statuses: BTreeMap::default(),
+            server_version_num: None,
```

Comment:
@abonander I can only put these things in `PgStream` instead of `PgConnection` without refactoring the Postgres part
and there is no mechanism to mark the connection as broken and not be used in the future when receiving invalid server message (ex. client_encoding change to non `UTF-8`) . Any suggestion or plan?

Reply:
Considering we were just ignoring `ParameterStatus` before, I don't think it matters that much. Maybe in the future with this data we can add support for non-UTF-8 text encodings but those are also rapidly falling out of use, so probably only if someone comes along and says "I really really need support for this text encoding and I'm willing to pay a bug bounty for it."

I'm also fine having this stuff on `PgStream`, it's easier than refactoring all calls of `.recv()` to handle `ParameterStatus`.